---
title: "SCN Proteomics MetaboAnalystR Pipeline"
author: "Theresa Alexander"
date: "`r Sys.Date()`"
output:
  html_document:
    code_download: true
    code_folding: show
    fig_caption: true
    fig_height: 7
    fig_width: 7
    highlight: tango
    keep_md: false
    mode: selfcontained
    number_sections: true
    self_contained: true
    theme: readable
    toc: true
    toc_float:
      collapsed: false
      smooth_scroll: false
  rmdformats::readthedown:
    code_download: true
    code_folding: show
    df_print: paged
    fig_caption: true
    fig_height: 7
    fig_width: 7
    highlight: tango
    width: 300
    keep_md: false
    mode: selfcontained
    toc_float: true
  BiocStyle::html_document:
    code_download: true
    code_folding: show
    fig_caption: true
    fig_height: 7
    fig_width: 7
    highlight: tango
    keep_md: false
    mode: selfcontained
    toc_float: true
---
<style type="text/css">
body, td {
  font-size: 16px;
}
code.r{
  font-size: 16px;
}
pre {
 font-size: 16px
}
</style>

# Load Libraries

These are the packages used in this analysis. 
<br>

```{r, message = FALSE, warning = FALSE}
#install instructions found here: https://www.metaboanalyst.ca/docs/RTutorial.xhtml#1.2%20Installation
library("MetaboAnalystR")
library("stringr")
library("dplyr")
library("ggplot2")
library("gplots")
library("xlsx")
library("ggrepel")
library("httr")
library("plyr")
library("gprofiler2")
library("hpgltools")
library("corrplot")
library("RColorBrewer")
library("patchwork")
source("../helper_functions.R")
```


```{r, echo = FALSE}
colors <- c("FF9D27", "68A3FF")
my_colors <- c("#fa00ed", "#d400c9", "#a600a6", "#9e0096", "#690063", "#260024", "#000000", "#1f1e00", "#474500", "#7d7900", "#bab400", "#e8e100", "#fff700")

my_colors <- c("#ff00ff", "#ed00ed", "#d600d6", "#c200c2", "#a600a6", "#820082", "#660066", "#4a004a", "#2e002e", "#140014", "#000000", "#1b1c00", "#2c2e00", "#414500", "#656b00", "#788000", "#949e00", "#a7b300", "#c1cf00", "#d8e800", "#edff00")
```

# Load Data and Clean

This metaboanalyst pipeline follows the vignette for data processing, DE, and GSEA. We first initialize an mSet, read in the data, and use the SanityCheckData() function to compute missingness, number of groups, and other sanity checks.

```{r}
prot_label_mapped <- read.xlsx("datasets/TotalProteinlist.xlsx", sheetIndex = 1)
```


<br>
```{r}
mSet<-InitDataObjects("conc", "stat", FALSE)
mSet<-Read.TextData(mSet, "datasets/TotalProteinlist_metaboanalyst.csv", "colu", "disc")
mSet<-SanityCheckData(mSet)
```



# Normalization and Filtering

We need to now normalize. We will use quantile normalization for row (protein across samples) normalization, will log_10 the values, and mean center. 

```{r}
mSet<-ReplaceMin(mSet) # impute 0 values to remove missing values with 1/5 of the min positive values for their corresponding samples (MetaboanalsytR's default imputation method)
mSet<-FilterVariable(mSet, "none", "F", 25)
mSet<-PreparePrenormData(mSet)
mSet<-Normalization(mSet, rowNorm = "QuantileNorm", "LogNorm", "MeanCenter", ratio=FALSE, ratioNum=20) #normalize and scale data
mSet<-PlotNormSummary(mSet, "norm_0_", "png", 72, width=NA)
mSet<-PlotSampleNormSummary(mSet, "snorm_0_", "png", 72, width=NA)
```


# PCA calculation and plotting

We will visualize global proteomic similarities/differences by way of PCA and visualize both the PC loadings and scores using ggplot. 

```{r}
mSet<-PCA.Anal(mSet)
mSet<-PlotPCAPairSummary(mSet, "pca_pair_0_", "png", 72, width=NA, 5)
mSet<-PlotPCAScree(mSet, "pca_scree_0_", "png", 72, width=NA, 5)
mSet<-PlotPCA2DScore(mSet, "pca_score2d_0_", "png", 72, width=NA, 1,2,0.95,0,0)
mSet<-PlotPCALoading(mSet, "pca_loading_0_", "png", 72, width=NA, 1,2);
mSet<-PlotPCABiplot(mSet, "pca_biplot_0_", "png", 72, width=NA, 1,2)
mSet<-PlotPCA3DLoading(mSet, "pca_loading3d_0_", "json", 1,2,3)
mSet<-PlotPCAPairSummary(mSet, "pca_pair_1_", "png", 72, width=NA, 2)

pca_loadings <- read.csv2(file = "pca_loadings.csv", sep = ",")
pca_score <- read.csv2(file = "pca_score.csv", sep = ",")
pca_score[,-c(1)] <- sapply(pca_score[,-c(1)], as.numeric)
pca_score$Time <- c("P08", "P08", "P08", "P08", "P08", "P21", "P21", "P21", "P21")
  
#set confidence interval threshold as well as alpha level for plotting the ellipses
ci_95 = .95
alpha = .3

pca <- ggplot(pca_score, aes(x = PC1, y = PC2, col = Time)) + 
  geom_point(size = 3) +
  theme_classic() +
  xlab(paste0("PC1: ", round(mSet$analSet$pca$variance[1],4)*100, "% variance")) +
  ylab(paste0("PC2: ", round(mSet$analSet$pca$variance[2],4)*100, "% variance")) +
  #ggtitle("PCA of Normalized Total Protein", subtitle = "CI = 95%") +
  stat_ellipse(mapping = aes(group = Time, 
                             fill = Time),
                              geom = "polygon", type = "t", level = ci_95, alpha = alpha) +
  scale_color_manual(values=c("#FF9D27", "#68A3FF"))  + 
  scale_fill_manual(values=c("#FF9D27", "#68A3FF")) + 
  theme(axis.text=element_text(size=12),
        axis.title=element_text(size=16),
        legend.text=element_text(size=10),
        legend.title = element_text(size=14),
        title = element_text(size=16),
        legend.position = "none") 

pca
```
<br>

Globally, across all the proteins identified and quantified, we see differences between the populations in the P8 vs the P21 samples.

<br>
To look into the specific differences between these samples, we will do a differential expression analysis using MetaboanalsytR's standard pipeline.
<br>

# DE

We first compute the fold change between P8 and P21 for each protein, then compute a FDR adjusted p-value for each protein.
<br>

```{r}
mSet <- FC.Anal(mSet, 2.0, 1)
mSet <- PlotFC(mSet, "fc_0_", "png", 72, width=NA)

mSet <- Ttests.Anal(mSet, threshp = 0.05, paired = FALSE)
mSet <- PlotTT(mSet, "tt_0_", "png", 72, width=NA)
```

<br>
Now, we can combine the fold change results and the t-test results to create a volcano plot to visualize the DE results.

```{r}
volc_df <- as.data.frame(cbind("accession_id" = names(mSet$analSet$tt$p.log), 
                               "genename" = prot_label_mapped$Gene.Name,
                               "adj.p" = mSet$analSet$tt$p.value,
                               "log.adj.p" = mSet$analSet$tt$p.log,
                               "log2_FC" = mSet$analSet$fc$fc.log)) #create data frame which combines all the needed information to plot

volc_df$log.adj.p <- as.numeric(volc_df$log.adj.p)
volc_df$log2_FC <- as.numeric(volc_df$log2_FC)
volc_df$adj.p <- as.numeric(volc_df$adj.p)

volc_df$Significance <- NA
volc_df[volc_df$log2_FC >= 1  & volc_df$adj.p <= .05, ][["Significance"]] <- "P21 Enriched"
volc_df[volc_df$log2_FC <= -1  & volc_df$adj.p <= .05, ][["Significance"]] <- "P08 Enriched"
volc_df[abs(volc_df$log2_FC) < 1 | volc_df$adj.p > .05, "Significance"] <- "Not Enriched"

my_y_lab <- bquote(paste("-Log"[10],"(", italic("p"), ")"))
my_x_lab <- bquote(paste("-Log"[2],"(P21/P8 Fold Change)"))

p <- ggplot(volc_df, aes(x = log2_FC, y = log.adj.p, label = genename)) + 
  geom_point(aes(colour = Significance)) + 
  geom_vline(xintercept = c(-1,1)) +
  geom_hline(yintercept = -log10(.05)) +
  theme_classic() +
  labs(x = my_x_lab, y = my_y_lab) +
  theme(legend.position="none") +
  scale_color_manual(values=c('Grey',"#FF9D27", "#68A3FF"))
```


```{r}
genes <- c("Sparcl1", "Plxnc1", #"Gabarapl2", "Dlg1", "Ncam2", "Ykt6", 
           "Dcx", #"Astn1", "Efnb3", "Pclo", 
           "Syap1", #"Ptn", "Daam1", 
           #"Cadm4", "Lmnb1", "Rtn4", #"Nsg2", "Dpysl5", 
           "Nectin1", #"Map1b", 
           #"Ctnna2", "Ntm", "Stx1a", "Thy1", "Cmtm5",
           "Syn1", #"Slc6a11", "Gad2", 
           #"Adam23", 
           "Sv2a", "Tnr", #"Ptprn2", "Ctsb", "Gripap1", "Slc17a6", 
           "Igsf8", 
           "Mbp",
           "Plp1", 
           #"Slc25a46",
           "Dpysl3")

volcano_plot <- p + geom_label_repel(data=filter(volc_df, (genename %in% genes)), nudge_x = -.1, 
                     nudge_y = .45, max.overlaps =20) + 
  theme(axis.text=element_text(size=12),
        axis.title=element_text(size=16),
        legend.text=element_text(size=14),
        legend.title = element_text(size=20),
        title = element_text(size=16),
        legend.position = "none") + 
  guides(colour = guide_legend(override.aes = list(size=6)))


volcano_plot

#ggsave(volcano_plot, filename = "volcanoplot.eps", width = 5, height = 7, device = cairo_ps)
```

# PCA Loadings

Adding colors to the PCA plot which correspond to the significant DE proteins.

```{r}
pca_loadings <- as.data.frame(cbind(pca_loadings, 
                                    "Significance" = volc_df$Significance,
                                    "accession_id" = names(mSet$analSet$tt$p.log), 
                                    "genename" = prot_label_mapped$Gene.Name))

pca_loads_plot <- ggplot(pca_loadings %>% arrange(Significance), aes(x = as.numeric(PC1), y = as.numeric(PC2), 
                                                          color = Significance, 
                                                          label = genename)) + 
  geom_point(size = 3) +
  theme_classic() +
  scale_color_manual(values=c('#F5F5F5',"#FF9D27", "#68A3FF"))

pca_loads_plot <- pca_loads_plot  + geom_label_repel(data=filter(pca_loadings, (genename %in% genes)), 
                        max.overlaps =20,
                        color = "black") +
  labs(x = "PC Loadings 1", y = "PC Loadings 2") +
  theme(axis.text=element_text(size=12),
        axis.title=element_text(size=16),
        legend.text=element_text(size=10),
        legend.title = element_text(size=14),
        title = element_text(size=16),
        legend.position = "none") 

pca_loads_plot
```


# Plot Heatmap
<br>
To visualize the proteomic abundances in each sample for the signifcantly DE proteins, we can create a heatmap. We will use thresholds of FDR adjusted p-value <= .05 and log2 fold change > 1 as significance cutoffs. <br>


```{r}
volc_df_sig <- volc_df %>%
  filter(adj.p <= .05 & abs(log2_FC) >= 1) %>%
  arrange(dplyr::desc(abs(log2_FC))) %>%
  head(n=200)

keepers <- colnames(mSet$dataSet$norm)[colnames(mSet$dataSet$norm) %in% volc_df_sig$accession_id]

norm_subset <- mSet$dataSet$norm[, keepers] %>% #subset significant DE genes list
  t() %>%
  head(n=200) %>%
  as.matrix()

colnames(norm_subset) <- c("P8 Bio Rep 1", "P8 Bio Rep 2", "P8 Bio Rep 3", "P8 Bio Rep 4", "P8 Bio Rep 5", "P21 Bio Rep 1", "P21 Bio Rep 2", "P21 Bio Rep 3", "P21 Bio Rep 4")
```


```{r}
my_colors <- brewer.pal(9, "Greys")
#setEPS()
#postscript("figures/DE_heatmap_bluescale.eps")
heatmap.2(norm_subset, 
          trace='none', 
          dendrogram=c("col"), 
          scale = "row", 
          col= my_colors, 
          #cexCol = 1.5, 
          #cexRow = 1.5,  
          main = "Normalized Protein Abundance", 
          labRow = FALSE, 
          srtCol=45,  
          adjCol = c(1,1), 
          ylab = "DE Proteins", 
          #margins = c(8, 3),
          density.info="none",
          ColSideColors = c(rep("#FF9D27", 5), rep("#68A3FF", 4))) #create a heatmap of the DE genes
#dev.off()
```

# Correlation Between Samples Using DE Genes
<br>
To visualize the the similarity between the expression of these DE proteins in the P8 and P21 samples, we can compute the correlation in their expression profiles and use hierarchical clustering of the samples. 
<br>
```{r}
my_colors <- brewer.pal(8, "Greys")
cor_samples <- norm_subset %>%
  cor()

#setEPS()
#postscript("figures/corrplot_DEgenes_greyscale.eps")
corrplot(cor_samples, method="circle", order="hclust", 
         tl.col = c(rep("#68A3FF", 4), 
                    rep("#FF9D27", 5)),
          col = my_colors,
          tl.cex = 1.25,
         addrect = 2)
#dev.off()
```

# Creat Input for Pathview

We would like to visualize the pathways which are enriched in these DE results. To do that, we will use the R package pathview. We first need to create a cleaned datasets input to pathview with ensembl IDs as the Id input.

```{r}
pv_df <- volc_df %>%
  filter(adj.p <= .05)

convert <- gconvert(query = pv_df[["genename"]], organism = "mmusculus", 
                                            target="ENSG", 
                                            mthreshold = Inf, 
                                            filter_na = TRUE) %>%
  select("input", "target")

pv_df_merged <- merge(pv_df, convert, by.x = "genename", by.y = "input") 
pathview_df <- pv_df_merged %>% 
  select("target", "log2_FC")
```




## Write DE Datasets to files 

This will create 3 different files.
<br>
1. `ipRGC_proteomics_sigDE.xlsx`, which has the significant proteins from the metaboanalystR DE  <br>
2. `ipRGC_proteomics_DE_all.xlsx`, which has all the proteins with associated p-values and fod changes from the metaboanalystR DE <br>
3. `ipRGC_proteomics_pathview_input.csv`, which has the information needed as input to pathview to create KEGG pathway visualizations <br>


```{r, eval = FALSE}
desc <- prot_int_old %>%
  select("Protein.Name", "Accession.ID")

write.xlsx(volc_df_sig, file = "datasets/ipRGC_proteomics_sigDE.xlsx", sheetName = "P8_vs_P21") #write significant DE results
volc_df_desc <- merge(volc_df, desc, by.x = "accession_id", by.y = "Accession.ID") #add protein descriptions as field in the DE results to write to xlsx
write.xlsx(volc_df_desc, file = "datasets/DEres_ALL_descriptions.xlsx", sheetName = "P08vP21") #write all DE results w/ description
write.csv(pathview_df, file =  "datasets/ipRGC_proteomics_pathview_input.csv", quote = FALSE) #write pathview input sheet
```


# Gene Set Enrichment Analysis

We would like to determine which gene ontology terms are represented in our DE results. To do this, we will use gprofiler2 (along with an aider package called hpgltools) to do this GSEA analysis. 

```{r}
DEres_down <- volc_df %>%
  filter(adj.p <= .05 & log2_FC < -1) %>%
  arrange(dplyr::desc(abs(log2_FC)))
DEres_up <- volc_df %>%
  filter(adj.p <= .05 & log2_FC > 1) %>%
  arrange(dplyr::desc(abs(log2_FC)))

gprofiler_down <- gost(query = DEres_down$genename, 
                organism = "mmusculus", 
                evcodes = TRUE)

gprofiler_down[["go"]] <- gprofiler_down$result %>%
  filter(source %in% c("GO:CC", "GO:BP", "GO:MF")) %>%
  filter(p_value <= 0.05)

gprofiler_up <-  gost(query = DEres_up$genename, 
                organism = "mmusculus", 
                evcodes = TRUE)
gprofiler_up[["go"]] <- gprofiler_up$result %>%
  filter(source %in% c("GO:CC", "GO:BP", "GO:MF")) %>%
  filter(p_value <= 0.05) 
```

## GSEA results

These are the top 30 GO terms for both P8 and P21 enriched DE proteins.

```{r}
gprofiler_down$go[order(gprofiler_down$go$recall, decreasing = TRUE), ] %>%
  select("term_name", "recall", "p_value", "intersection_size") %>%
  head(30) %>%
  knitr::kable("simple", row.names = FALSE, caption = "P8 Enriched GO Terms")

gprofiler_up$go[order(gprofiler_up$go$recall, decreasing = TRUE), ] %>%
  select("term_name", "recall", "p_value", "intersection_size") %>%
  head(30) %>%
  knitr::kable("simple", row.names = FALSE, caption = "P21 Enriched GO Terms")
```


## Write the GSEA results to an excel file

```{r}
gprofiler_down[["go"]] %>%
   select("term_name", "recall", "p_value", "intersection_size", "intersection") %>%
  write.xlsx(file = "datasets/SCNProteomics_GSEA_2022_P8.xlsx", sheetName = "P8")

gprofiler_up[["go"]] %>%
   select("term_name", "recall", "p_value", "intersection_size", "intersection") %>%
  write.xlsx(file = "datasets/SCNProteomics_GSEA_2022.xlsx", sheetName = "P21")
```



```{r}
sessionInfo()
```
